From ed1c15149d58977bd4a4be07b1e1234eca3d5655 Mon Sep 17 00:00:00 2001
From: Marc Mutz <marc.mutz@qt.io>
Date: Thu, 11 Sep 2025 11:34:25 +0200
Subject: [PATCH 148/553] tst_selftests: make generate.sh work in checkouts
 with name != qtbase

The script checked that a certain path element is 'qtbase', which
fails if the qtbase checkout is called something else. E.g., for me,
that's qtbase-submit.

Fix by checking the other path elements, but not the module name.

Amends 3ac029f674edea1cc91014345f445a9a8d1fc89f.

Pick-to: 6.9 6.8 6.5
Change-Id: I651cb168990b93939f339f8d75cb71c31fa6f2a4
Reviewed-by: Ivan Solovev <ivan.solovev@qt.io>
(cherry picked from commit ee5aa5a32b48260416a3edcbff6be6c271cc9684)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 tests/auto/testlib/selftests/generate_expected_output.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/tests/auto/testlib/selftests/generate_expected_output.py b/tests/auto/testlib/selftests/generate_expected_output.py
index b1d6ec86ccc..220fb4b1113 100755
--- a/tests/auto/testlib/selftests/generate_expected_output.py
+++ b/tests/auto/testlib/selftests/generate_expected_output.py
@@ -131,7 +131,11 @@ class Cleaner (object):
         scriptPath = os.path.dirname(os.path.abspath(__file__))
         hereNames, depth = scriptPath.split(os.path.sep), 5
         hereNames = hereNames[-depth:] # path components from qtbase down
-        assert hereNames[0] == 'qtbase', ('Script moved: please correct depth', hereNames)
+        # don't check hereNames[0]: the qtbase checkout might not be named 'qtbase'
+        assert hereNames[1] == 'tests', ('Script moved: please correct depth', hereNames)
+        assert hereNames[2] == 'auto', ('Script moved: please correct depth', hereNames)
+        assert hereNames[3] == 'testlib', ('Script moved: please correct depth', hereNames)
+        assert hereNames[4] == 'selftests', ('Script moved: please correct depth', hereNames)
         qtbase_dir = os.path.realpath(os.path.join(scriptPath, '..', '..', '..', '..'))
         qtver = Cleaner._read_qt_version(qtbase_dir)
         hereNames = tuple(hereNames)
-- 
2.51.2

